xcm/xcm-executor/integration-tests/src/lib.rs

```diff
@@ -71,6 +72,62 @@ fn basic_buy_fees_message_executes() {
 	});
 }
 
+#[test]
+fn transact_recursion_limit_works() {
+	sp_tracing::try_init_simple();
+	let mut client = TestClientBuilder::new()
+		.set_execution_strategy(ExecutionStrategy::AlwaysWasm)
+		.build();
+
+	let mut msg = Xcm(vec![ClearOrigin]);
```

Comment:
only one of these two needs to be stateful - the other can be built in the loop each iteration.

Reply:
```suggestion
```